File: server/src/main/java/com/fluffy/oauth2/infra/google/dto/GoogleAccessTokenRequest.java
package com.fluffy.oauth2.infra.google.dto;

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;

@JsonNaming(SnakeCaseStrategy.class)
public record GoogleAccessTokenRequest(
        String code,
        String clientId,
        String clientSecret,
        String grantType,
        String redirectUri
) {
}

File: server/src/main/kotlin/com/fluffy/oauth2/infra/google/request/GoogleAccessTokenRequest.kt
package com.fluffy.oauth2.infra.google.request

import com.fasterxml.jackson.databind.PropertyNamingStrategies.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming

@JsonNaming(SnakeCaseStrategy::class)
data class GoogleAccessTokenRequest(
    val code: String,
    val clientId: String,
    val clientSecret: String,
    val grantType: String,
    val redirectUri: String,
)
